secure/storage/vault/src/lib.rs

```diff
@@ -366,20 +397,26 @@ impl Client {
 
     /// Create or update a key/value pair in a given secret store.
     pub fn write_secret(&self, secret: &str, key: &str, value: &str) -> Result<(), Error> {
-        let request = ureq::put(&format!("{}/v1/secret/data/{}", self.host, secret));
+        let request = self
+            .agent
+            .put(&format!("{}/v1/secret/data/{}", self.host, secret));
         let resp = self
             .upgrade_request(request)
             .send_json(json!({ "data": { key: value } }));
         match resp.status() {
-            200 => Ok(()),
+            200 => {
+                // explicitly read contents so the socket can be re-used
+                resp.into_string()?;
+                Ok(())
+            }
```

Comment:
@aching , @JoshLind , @zekun000 this is the other case... we'd still be creating a new connection each proposal since the writes weren't flushing the buffer... this actually feels like a weird issue on the ureq side, and I'm going to file an issue.

Reply:
Wow, yeah, this seems odd to me..

Reply:
it probably should impl Drop for this

Reply:
https://github.com/algesten/ureq/issues/94

Reply:
Curious why this isn't necessary for explicitly reading contents for other responses - such as rotate_key, sign_ed25519, etc.?

Reply:
Well, I wouldn't say it isn't necessary, these aren't used in heavily used loops so the connection churning is extremely unlikely. Though as good hygiene, I'll add it!